### ISSUE TITLE
Multishot and how % modifiers calculate damage

### ISSUE THREAD

**Issue by watchyoursixx** [watchyoursixx/classic-hunter#4]
Does the % bonuses such as Barrage increase the multishot damage bonus AND multi shot formula overall, or just multi shot bonus or just multi shot formula overall?

**Comment by Zarant** [comment]
Just adding as reference this forum post from 2006, it has all the combat formulas:
https://web.archive.org/web/20150425074901/http://forums.elitistjerks.com/topic/8267-hunter-damage-cycles-breakdown/

> FORMULAE USED
> Auto Shot damage: SlayingMod * (AmmoDPS * WeaponSpeed+RWSMod * (RAP/14 * WeaponSpeed+Scope+AverageWeaponDamage))
> Aimed Shot damage: SlayingMod * (AmmoDPS * WeaponSpeed+RWSMod * (RAP/14 * 2.8+Scope+AverageWeaponDamage+AimedShotBonus))
> Multi-Shot damage: BarrageMod * GSMod * SlayingMod * MSCoEff * 
 (AmmoDPS * WeaponSpeed+RWSMod * (RAP/14 * 2.8+Scope+AverageWeaponDamage+MultiShotBonus))
> 
> CritDamage (all shots): HitDamage * (2+MortalShotsMod) * SlayingMod
> 
> Factoring in crits (all shots): HitChance * HitDamage + CritChance * Critdamage
> 
> AimedShotBonus = 600
> MultiShotBonus = 150
> MortalShotsMod = 0.30
> SlayingMod = 1.03 if Slaying Talent, 1 if not
> RWSMod = 1.05 if MM spec, 1 if LR spec
> BarrageMod = 1.15 if MM spec, 1 if LR spec
> GSMod = 1.15 if 8/8 Giantstalker bonus, 1 if not

In order to test that properly I suggest using fixed damage weapons like the white crossbows sold in Stormwind or Undercity. (https://classic.wowhead.com/item=15808/fine-light-crossbow)

**Comment by watchyoursixx** [comment]
Confirmed on the beta build 30901.
Using the log [here](https://github.com/watchyoursixx/classic-hunter/files/3325544/WoWCombatLog.bds.lightxbow.tests.txt) I was able to determine the formula for multi-shot and the interaction with barrage/RWS.

With 402 RAP, 13 DPS arrows, 40 bonus damage from MS, RWS, and barrage, MS did 211 damage.
The formula for multi shot observed would then be **(13 * 2.7 + (2.8 * 402/14 + 40 + 20)) * 1.15 * 1.05** giving us the value of 211 rounded down and 212 rounded up. In the tests bds tried with and without the talents. With no talents Multishot hit for 175, and with just barrage multishot hit for 201.

So based on these calculations, no the bonus is not multiplied by barrage, only the full formula. Also proves that the EJ post was slightly off in the damage calculation as RWS is multiplied by the whole formula instead of excluding the arrow dps.
